app/src/main/java/ch/hikemate/app/ui/components/BackButton.kt

```diff
@@ -34,8 +39,18 @@ fun BackButton(
     modifier: Modifier = Modifier,
     onClick: () -> Unit = { navigationActions.goBack() }
 ) {
+
+  var wantsToClick by remember { mutableStateOf(false) }
+
+  LaunchedEffect(wantsToClick) {
```

Comment:
In the Run Hike screen, the `wantToNavigateBack` variable is still used by this button. Just to make you aware, it does not need to be changed but might be nice.

Reply:
Actually, the RunHikeScreen has multiple paths that lead to going back, so this makes sense! (Not only the BackButton click)